python/GangaLSST/Lib/Im3ShapeApp/Im3ShapeApp.py

```diff
@@ -0,0 +1,117 @@
+        'run_dir': SimpleItem(defvalue='im3shape-grid', types=[str], doc="Directory on the WN where the binary is"),
+        ## Below is needed for prepared state stuff
+        'is_prepared': SimpleItem(defvalue=None, strict_sequence=0, visitable=1, copyable=1, hidden=0, typelist=[None, bool, ShareDir], protected=0, comparable=1, doc='Location of shared resources. Presence of this attribute implies the application has been prepared.'),
+        'hash': SimpleItem(defvalue=None, typelist=[None, str], hidden=0, doc='MD5 hash of the string representation of applications preparable attributes'),
+    })
+    _category = 'applications'
+    _name = 'Im3ShapeApp'
+    _exportmethods = ['prepare', 'unprepare']
+
+    def unprepare(self, force=False):
+        """
+        Revert an Im3ShapeApp application back to it's unprepared state.
+        args:
+            force (bool): should force the unprepare step to run
+        """
+        logger.debug('Running unprepare in Im3ShapeApp')
+        if self.is_prepared is not None:
+            self.decrementShareCounter(self.is_prepared.name)
+            self.is_prepared = None
+        self.hash = None
+
+    def prepare(self, force=False):
+        """
+        This prepares the Im3ShapeApp application and copies any LocalFile objects which are allocated to: im3_location, ini_location and blacklist into the prepared sandbox to be shipped to the WN
+        Args:
+            force (bool): Should force the prepare step to run
+        """
+
+        if (self.is_prepared is not None) and (force is not True):
+            raise ApplicationPrepareError('%s application has already been prepared. Use prepare(force=True) to prepare again.' % getName(self))
+
+        # lets use the same criteria as the configure() method for checking file existence & sanity
+        # this will bail us out of prepare if there's somthing odd with the job config - like the executable
+        # file is unspecified, has a space or is a relative path
+        self.configure(self)
+        logger.info('Preparing %s application.' % getName(self))
+        setattr(self, 'is_prepared', ShareDir())
+        logger.info('Created shared directory: %s' % (self.is_prepared.name))
+
+        try:
+            # copy any 'preparable' objects into the shared directory
+            send_to_sharedir = self.copyPreparables()
+            # add the newly created shared directory into the metadata system
+            # if the app is associated with a persisted object
+            self.checkPreparedHasParent(self)
+            for file_ in [self.ini_location, self.im3_location, self.blacklist]:
+                if isinstance(file_, LocalFile):
+                    self.copyIntoPrepDir(file_.namePattern)
+                assert type(file_) in [LocalFile, DiracFile]
+            # return
+            # [os.path.join(self.is_prepared.name,os.path.basename(send_to_sharedir))]
+            self.post_prepare()
+
+        except Exception as err:
```

Comment:
We shouldn't be catching all exceptions here if we can avoid it. `Exception` is dangerous as it includes `NameError` etc. which hide typos.